simulator/src/test/java/com/hedera/block/simulator/grpc/PublishStreamGrpcClientImplTest.java

```diff
@@ -77,6 +79,28 @@ void streamBlock() {
         assertTrue(result1);
     }
 
+    @Test
+    void streamBlockReturnsFalse() throws NoSuchFieldException, IllegalAccessException {
+        BlockItem blockItem = BlockItem.newBuilder().build();
+        Block block = Block.newBuilder().items(blockItem).build();
+
+        PublishStreamGrpcClientImpl publishStreamGrpcClient =
+                new PublishStreamGrpcClientImpl(grpcConfig, blockStreamConfig);
+        publishStreamGrpcClient.init();
+        Field allowNextField = PublishStreamGrpcClientImpl.class.getDeclaredField("allowNext");
+
+        try {
+            allowNextField.setAccessible(true);
+            AtomicBoolean allowNext = (AtomicBoolean) allowNextField.get(publishStreamGrpcClient);
+            allowNext.set(false);
+        } finally {
+            allowNextField.setAccessible(false);
+        }
+
+        boolean result = publishStreamGrpcClient.streamBlock(block);
+        assertFalse(result);
+    }
```

Comment:
About the name of the variable, i agree, changed it.
I've also refactored, by adding it in the constructor, alothough I really dislike it, because it feels like I'm altering and introducing coupling in production code, just for the sake of easier testing. Tried different strategies and none feels right for testing, therefore injected in constructor is the way for now.

Reply:
Injection is the preferred mechanism for things that need to modified when unit testing, but yes, injecting an atomic boolean is not ideal.
It's most likely just a short-term quirk, however. I suspect that we'll eventually replace this single value with something like a stream control object in future work.